static/js/util/util.js

```diff
@@ -458,3 +458,41 @@ export function getUserDisplayName(profile: Profile): string {
 
   return `${first} ${last}${preferred_name}`;
 }
+
+const MAX_WIDTH = 512;
+const MAX_HEIGHT = 512;
+
+export const shrinkImage = (image: File, saveCallback: Function) => {
+  const fr = new FileReader();
+
+  fr.onload = () => {
+    const img = new Image();
+    img.src = fr.result;
+
+    let width = img.width;
+    let height = img.height;
+
+    if (width > height) {
+      if (width > MAX_WIDTH) {
+        height *= MAX_WIDTH / width;
+        width = MAX_WIDTH;
+      }
+    } else {
+      if (height > MAX_HEIGHT) {
+        width *= MAX_HEIGHT / height;
+        height = MAX_HEIGHT;
+      }
+    }
+
+    const canvas = document.createElement('canvas');
+    canvas.width = width;
+    canvas.height = height;
+
+    const ctx: any = canvas.getContext("2d");
+    ctx.drawImage(img, 0, 0, width, height);
+
+    return canvas.toBlob(saveCallback, 'image/jpeg');
```

Comment:
We want to use JPEG because it is a substantial reduction in size over PNG. I merged a change yesterday which switched that over.